### PR TITLE
add missed case for float types

### PR DESCRIPTION
**Pull request by borgerli**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
If we add fields of float32(or float64) type to some API type, it will report error of "unhandled type float32 in field xxx" when generating api testing compatibility data. This PR is to fix this problem.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by borgerli** [comment]
@liggitt  I'm not sure whether float types were intentionally ignored. Please review, thanks.

**Comment by liggitt** [comment]
floats are discouraged in API types (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) and often have issues perfectly round-tripping

is there a specific type you are encountering issues with you can link to?

**Comment by borgerli** [comment]
> floats are discouraged in API types (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) and often have issues perfectly round-tripping
> 
> is there a specific type you are encountering issues with you can link to?

Yes, we extended some API Resource by adding a field named `qps` of float32 type in version 1.22. When porting to 1.24, we ran into problem.

File: staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/construct.go [hunk 1]
```diff
@@ -173,6 +173,10 @@ func fill(dataString string, dataInt int, t reflect.Type, v reflect.Value, fillF
 		// use Convert to set into int alias types and different int widths correctly
 		v.Set(reflect.ValueOf(dataInt).Convert(t))
 
+	case reflect.Float32, reflect.Float64:
+		// use Convert to set into float types
+		v.Set(reflect.ValueOf(dataInt).Convert(t))
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/construct.go:178]: make sure this exercises the float round tripping with something like this:

```suggestion
		v.Set(reflect.ValueOf(float32(dataInt) + 0.5).Convert(t))
```

Comment (borgerli) [reply staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/construct.go]: updated code as sugguesed. thanks.

**Comment by pacoxu** [comment]
TestLegacyFallbackNoCache is a flake https://github.com/kubernetes/kubernetes/issues/115983

**Pull request merged**